src/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.cs

```diff
@@ -1117,6 +1117,108 @@ public AssemblyReference GetAssemblyReference(AssemblyReferenceHandle handle)
             return new AssemblyReference(this, handle.Value);
         }
 
+        public AssemblyName GetAssemblyName()
+        {
+            StringHandle nameHandle = AssemblyTable.GetName();
+
+            string name = (!nameHandle.IsNil) ? GetString(nameHandle) : null;
+
+            Version version = AssemblyTable.GetVersion();
+
+            StringHandle cultureHandle = AssemblyTable.GetCulture();
+            string cultureName = (!cultureHandle.IsNil) ? GetString(cultureHandle) : null;
+
+            BlobHandle publicKeyHandle = AssemblyTable.GetPublicKey();
+            byte[] publicKeyOrToken = !publicKeyHandle.IsNil ? GetBlobBytes(publicKeyHandle) : null;
+
+            var assemblyName = new AssemblyName(name)
+            {
+                Version = version,
+                CultureName = cultureName
+            };
+
+            assemblyName.SetPublicKey(publicKeyOrToken);
+            SetAssemblyNameFlags(assemblyName);
+            SetAssemblyNameHashAlgorithm(assemblyName);
+
+            return assemblyName;
+        }
+
+        private void SetAssemblyNameFlags(AssemblyName assemblyName)
```

Comment:
@tmat Would it make sense to move this private method (SetAssemblyNameFlags) to AssemblyName and call it SetFlags() ? or is this ok here as a private method?